src/Lib/Managers/ItemsManager.vala

```diff
@@ -49,7 +49,41 @@ public class Akira.Lib.Managers.ItemsManager : Object {
         window.event_bus.change_item_z_index.connect (on_change_item_z_index);
     }
 
-    public Models.CanvasItem? insert_item (double x, double y) {
+    public void insert_image (Services.FileImageProvider provider) {
+        var selected_bound_manager = window.main_window.main_canvas.canvas.selected_bound_manager;
```

Comment:
Here I see the same issue we had with ItemsManager as per having a global SelectBoundManager and not one which is a field of the Canvas. 

I know we can defer this change to another PR, but I simply want to add some context to this choice by bringing this as an example.